### PR TITLE
Split platforms for clang in DEPS to prepare for autoroller

### PR DESCRIPTION
**Pull request by dnfield**
https://bugs.chromium.org/p/skia/issues/detail?id=11197#c3

### CODE REVIEW

**Review by jason-simmons — APPROVED** [review]

File: DEPS [hunk 1]
```diff
@@ -498,14 +498,25 @@ deps = {
     'dep_type': 'cipd',
   },
 
-  'src/buildtools/{host_os}-x64/clang': {
+  'src/buildtools/mac-x64/clang': {
```
Comment (zanderso) [inline DEPS:501]: Consider adding a comment explaining why they are separate.

**Review by zanderso — APPROVED** [review]

File: DEPS (hunk 1 above)
Comment (dnfield) [reply DEPS:503]: Done

**Comment by dnfield** [comment]
I'm ignoring the timeout and the failure because they're not related to this PR.

**Comment by dnfield** [comment]
Also landed on red to kick the tree, which is hung on an infra failure. This is a very safe patch since it is a no-op refactor.

**Pull request merged**